lib/matplotlib/axes/_secondary_axes.py

```diff
@@ -185,20 +182,17 @@ def _set_scale(self):
         if self._orientation == 'x':
             pscale = self._parent.xaxis.get_scale()
             set_scale = self.set_xscale
-        if self._orientation == 'y':
+        else:  # 'y'
             pscale = self._parent.yaxis.get_scale()
             set_scale = self.set_yscale
         if pscale == self._parentscale:
             return
 
-        if pscale == 'log':
-            defscale = 'functionlog'
-        else:
-            defscale = 'function'
-
         if self._ticks_set:
             ticks = self._axis.get_ticklocs()
 
+        defscale = 'functionlog' if pscale == 'log' else 'function'
+
         # need to invert the roles here for the ticks to line up.
         set_scale(defscale, functions=self._functions[::-1])
```

Comment:
Optional: IMHO, it's simpler in this case to inline than to do the extra indirection to store in a variable.

```suggestion
        # need to invert the roles here for the ticks to line up.
        set_scale('functionlog' if pscale == 'log' else 'function',
                  functions=self._functions[::-1])
```